src/ProjectOrigin.Vault/Jobs/PublishCheckForWithdrawnCertificatesCommandJob.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MassTransit;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using ProjectOrigin.Vault.CommandHandlers;
+using ProjectOrigin.Vault.Database;
+using ProjectOrigin.Vault.Options;
+
+namespace ProjectOrigin.Vault.Jobs;
+
+public class PublishCheckForWithdrawnCertificatesCommandJob : BackgroundService
+{
+    private readonly IBus _bus;
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly ILogger<PublishCheckForWithdrawnCertificatesCommandJob> _logger;
+    private readonly JobOptions _options;
+
+    public PublishCheckForWithdrawnCertificatesCommandJob(IBus bus, IUnitOfWork unitOfWork, IOptions<JobOptions> options, ILogger<PublishCheckForWithdrawnCertificatesCommandJob> logger)
+    {
+        _bus = bus;
+        _unitOfWork = unitOfWork;
+        _logger = logger;
+        _options = options.Value;
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            var lastExecutionTime = await _unitOfWork.JobExecutionRepository.GetLastExecutionTime(nameof(PublishCheckForWithdrawnCertificatesCommandJob));
```

Comment:
This logic has a high chance to be run by mutlipel pods at the same time. As far I remember we wanted to ensure exactly 1 only?

Reply:
yes that is what I'm trying to ensure with this logic